plugins/flytekit-k8s-pod/tests/test_pod.py

```diff
@@ -473,3 +474,32 @@ def dynamic_task_with_pod_subtask(dummy_input: str) -> str:
             assert dynamic_job_spec.tasks[0].k8s_pod.pod_spec["containers"][0]["resources"]["requests"]["gpu"] == "1"
 
     assert context_manager.FlyteContextManager.size() == 1
+
+
+def test_pod_config():
+    with pytest.raises(user.FlyteValidationException):
+        Pod(pod_spec=None)
+
+    with pytest.raises(user.FlyteValidationException):
+        Pod(pod_spec=V1PodSpec(containers=[]), primary_container_name=None)
+
+    selector = ({"node_group": "memory"},)
```

Comment:
just for the sake of correctness (e.g. in case someone copies this example), can you make this a dictionary?

Reply:
good catch. also great catch with the container name. I just changed that as well.